contracts/governance/extensions/GovernorOverrideDelegateVote.sol

```diff
@@ -0,0 +1,194 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {GovernorVotes} from "./GovernorVotes.sol";
+import {IVotes} from "../utils/IVotes.sol";
+import {IERC5805} from "../../interfaces/IERC5805.sol";
+import {SafeCast} from "../../utils/math/SafeCast.sol";
+import {Time} from "../../utils/types/Time.sol";
+import {VotesOverridable} from "../utils/VotesOverridable.sol";
+
+/**
+ * @dev Extension of {Governor} which enables delegatees to override the vote of their delegates. This module requires a
+ * token token that inherits `VotesOverridable`.
+ */
+abstract contract GovernorOverrideDelegateVote is GovernorVotes {
```

Comment:
Also, usually counting module does not inherit from GovernorVotes. Could we avoid that in heritance?

Reply:
It is both a voting module and a counting module. Maybe it could be something like `GovernorCountingAndVotesOverrideDelegate`?

Reply:
Yes it can be removed. The reason to inherit `GovernorVotes` is that this is strictly additional functionality to it and it makes inheriting `GovernorVotesQuorumFraction` simpler since it also inherits `GovernorVotes`.

Reply:
I agree it's not necessary to inherit from `GovernorVotes`, a developer building on Governor would pick a votes module and a counting module, while the quorum fraction is optional.

In such case, I would name it `GovernorCountingOverride`

Reply:
> I agree it's not necessary to inherit from `GovernorVotes`, a developer building on Governor would pick a votes module and a counting module, while the quorum fraction is optional.
> 
> In such case, I would name it `GovernorCountingOverride`

We could theoretically isolate this to be just a counting module and have a different voting module which implements `_getPastDelegate` and `_getPastBalanceOf`. This module would then be dependent on the usage of that voting module. Since they would be so intertwined and relatively useless on their own, I chose this approach which is a voting and counting module in one.

Reply:
I think inheriting from GovernorVotes to get acces to `token()` is fine.

For the name I think it should be succinct yet explicit about what it achieves. I think the current name may be a bit too long.

This is just a proposition:
```suggestion
abstract contract GovernorCountOverridable is GovernorVotes {
```